### PR TITLE
scripts: genpinctrl: add TSC pinctrl definition

### PR DESCRIPTION
**Pull request by arifbalik**
Added TSC regex to match and generate pin definitions. 

I'm not sure if I should also put here the new pinctrl files, but I've run the script and it generates the pins

### CODE REVIEW

File: scripts/genpinctrl/stm32-pinctrl-config.yaml [hunk 1]
```diff
@@ -299,3 +299,6 @@
 
 - name: JTAG PORT
   match: ^(SYS|DEBUG)_((JTMS-)?SWDIO|(JTCK-)?SWCLK|JTDI|JTDO(-TRACESWO|-SWO)?|(NJ)?JTRST)$
+
+- name: TSC
+  match: "^TSC_(?:G\\d+_IO\\d+|SYNC)$"
```
Comment (erwango) [inline scripts/genpinctrl/stm32-pinctrl-config.yaml:304]: Please sort alphabetically. Seems that the alphabetical sorting requires some review, but let's not making it worse.

**Review by erwango — COMMENTED** [review]
Thanks. Looks fine, but please split template changes from the new batch of generated files in a dedicated commit.

File: scripts/genpinctrl/stm32-pinctrl-config.yaml (hunk 1 above)
Comment (arifbalik) [reply scripts/genpinctrl/stm32-pinctrl-config.yaml:304]: Do you mean to move this definition somewhere in the file where `name` is in alphabetical order? or the generated pin definitions?

Comment (erwango) [reply scripts/genpinctrl/stm32-pinctrl-config.yaml:304]: > Do you mean to move this definition somewhere in the file where name is in alphabetical order? 

yup

Comment (arifbalik) [reply scripts/genpinctrl/stm32-pinctrl-config.yaml]: Got it, thanks

**Comment by arifbalik** [comment]
> @arifbalik I think @erwango meant split it into different commits, not into different PRs, anyway same result if we merge them in sequence, @erwango how do you want this?

Ah, my bad, I can put the commit here as well if you like

**Comment by fabiobaltieri** [comment]
> Ah, my bad, I can put the commit here as well if you like

If you don't mind, better to keep them under the same PR I think

**Review by fabiobaltieri — APPROVED** [review]

**Review by erwango — APPROVED** [review]
Thanks, I meant same PR but different commits indeed.
Next time, please put commits in logical order (template changes first, then new batch of files). Anyway not blocking, so "approved", thanks.

**Pull request merged**